[Joey Lee] iP

applepiofmyeye opened the pull request:
###Barbie Chatbot

> "Humans only have one ending. Ideas live forever." -- Ruth Handler ([source](https://collider.com/barbie-best-quotes-ranked/#quot-humans-only-have-one-ending-ideas-live-forever-quot)

Let your ideas live forever with this barbie chatbot. It's,

- Text-based
- ~FAST~ _SUPER_ FAST
- And Barbie approved!

All you need to do is, 
1. Download it 
2. Double click it
3. Add your tasks
4. Let it manage your tasks for you 😄 

And it's **FREE**!

Features:
- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Find tasks
- [ ] List tasks

If you're a Java user, you can use it to practise Java too! Here's the `main` method

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level 0

Commit:
Commit message for Tag-1

Commit:
Code recovery

Commit:
Level-2: Add, List

Commit:
Level-1: Echo

Commit:
Level-3: Mark as done, created new class Task

Commit:
Level-4: Todos, deadlines and barbie parties hehe

Commit:
automated UI Testing

Commit:
Level-5 Handle errors

Commit:
not the best.. but Level-6 v1

Commit:
A-Enums, using enums for commands, to inc readability

Commit:
Level-7 v1 -- manually saving, but also saving many different lists (but not being able to edit)

Commit:
Level-7

Commit:
Level-8: change to LocalDate and added to today's list at intro

Commit:
A-More-OOP: Adding more abstraction through types, storage, ui and utils class

Commit:
NOTE: working codegit add .! from Level-7 onwards may be a bit buggy until here

Commit:
A-Gradle: add gradle

applepiofmyeye committed:
Merge pull request #1 from applepiofmyeye/branch-org-packages

Merge A-packages with add-gradle-support

applepiofmyeye committed:
Merge pull request #2 from applepiofmyeye/branch-Level-7

Branch level 7

applepiofmyeye committed:
Merge pull request #3 from applepiofmyeye/branch-Level-8

Level-8: change to LocalDate and added to today's list at intro

applepiofmyeye committed:
Merge pull request #4 from applepiofmyeye/branch-more-OOP

A-More-OOP: Adding more abstraction through types, storage, ui and ut…

applepiofmyeye committed:
Merge pull request #5 from applepiofmyeye/branch-org-packages

NOTE: working codegit add .! from Level-7 onwards may be a bit buggy …

Commit:
A-JUnit: Add JUnit Tests

applepiofmyeye committed:
Merge pull request #6 from applepiofmyeye/branch-A-JUnit

A-JUnit increment

Commit:
A-JavaDoc: add JavaDoc

Commit:
A-CodingStandard: follow coding standards

Commit:
Level-9

javinchua commented on src/main/java/Barbie.java:
```diff
@@ -0,0 +1,182 @@
+import types.Party;
+import types.Task;
+import types.Todo;
+import exceptions.*;
```
It would be better to avoid wildcard imports, rather you could list out the imports of the classes that are being used.

javinchua commented on src/main/java/Barbie.java:
```diff
@@ -0,0 +1,182 @@
+    enum Command {
+        MARK,
+        UNMARK,
+        DEL,
```
I like that you have created an enumeration for the commands but perhaps it might be better to use the full word naming for the enum value of DELETE instead of DEL.

javinchua commented on src/main/java/Barbie.java:
```diff
@@ -0,0 +1,182 @@
+                                    }
+
+                                    desc = parts2[0];
+                                    LocalDate by = LocalDate.parse(parts2[1]);
```
Before accessing parts2[1], it would be good to ensure that parts2 actually has 2 elements, to avoid running into ArrayOutOfBoundsException.

javinchua commented on src/main/java/Barbie.java:
```diff
@@ -0,0 +1,182 @@
+
+                    Ui.barbie();
+
+                    switch (command) {
```
The main class, Barbie, seems to be handling many responsibilities. Consider using a command class to break down the responsibilities.

javinchua commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,122 @@
+                    String desc = taskParts[2];
+                    Task task;
+
+                    if (Objects.equals(taskType, "T")) {
+                        task = new Todo(desc);
+                    } else if (Objects.equals(taskType, "D")) {
+                        task = new Deadlines(desc, LocalDate.parse(taskParts[3]));
+                    } else if (Objects.equals(taskType, "P")) {
+                        task = new Party(desc, LocalDate.parse(taskParts[3]), LocalDate.parse(taskParts[4]));
```
```suggestion
                    if (String.equals(taskType, "T")) {
                        task = new Todo(desc);
                    } else if (String.equals(taskType, "D")) {
                        task = new Deadlines(desc, LocalDate.parse(taskParts[3]));
                    } else if (String.equals(taskType, "P")) {
                        task = new Party(desc, LocalDate.parse(taskParts[3]), LocalDate.parse(taskParts[4]));
```
Consider using string.equals for this scenario as it is more specific to the data types.

applepiofmyeye replied:
I didn't notice this, thanks!!

javinchua reviewed (commented):
Overall, I enjoy the abstraction that you have done to the Duke Class, especially in the Storage class, where you specifically remove particular lines depending on the command. An overall comment could be to add more Javadocs, especially for public methods as it will allow improved overall readability, specifically in the case of when u have overloaded methods, as this will allow the author and others to better understand the different scenarios for using the overloaded methods.

applepiofmyeye committed:
Merge pull request #7 from applepiofmyeye/branch-A-JavaDoc

A-JavaDoc: add JavaDoc

Commit:
Merge branch 'master' of https://github.com/applepiofmyeye/ip into branch-Level-9

Commit:
Merge changes from JavaDoc and CodingStandard branch

Commit:
Different todo list

applepiofmyeye committed:
Merge pull request #8 from applepiofmyeye/branch-A-CodingStandard

A-CodingStandard: follow coding standards

Commit:
Merge branch 'master' into branch-Level-9

Commit:
Merge Javadoc and CodingStandards

Commit:
Merge branch 'master' of https://github.com/applepiofmyeye/ip

Commit:
Merge Level-9 with master

applepiofmyeye committed:
Merge pull request #9 from applepiofmyeye/branch-Level-9

Level-9

Commit:
Fixed FIND command

applepiofmyeye committed:
Merge pull request #10 from applepiofmyeye/branch-Level-9

Fixed FIND command

Commit:
Comply to checkstyle

miljyy commented on src/main/java/Storage.java:
```diff
@@ -50,14 +51,15 @@ protected static void addToList(Path path, String desc) {
+
     /**
      * Writes to the file given, in the format of [type][status] [desc].
      * Eg. [T][ ] read book
```
I like how detailed your Javadoc is, even providing the example of the format that will be written to file.

miljyy commented on src/main/java/Barbie.java:
```diff
@@ -155,17 +138,32 @@ public static void main(String[] args) {
+                        list.add(indexNumber, new Task(input)); // Create a new Task
+                        indexNumber += 1; //Incrementing item counter
+                        // Output
+                        System.out.println("\t Okey Dokey! I've added this task into your list:\n"
+                                + "\t[ ] " + input);
```
Perhaps such output could be handled by Ui class?

miljyy reviewed (commented):
Overall I like how clean the code is, generally easy to read and understand!

applepiofmyeye committed:
Merge pull request #11 from applepiofmyeye/branch-A-Checkstyle

Comply to checkstyle

Commit:
Change exception handling format, add findTasks logic

Commit:
Add greater abstraction, add GUI functionality

applepiofmyeye committed:
Merge pull request #13 from applepiofmyeye/branch-Level-10

Add greater abstraction, add GUI functionality

Commit:
Add assertions for error handling

Commit:
Add javadocs, change long methods, checkstyle

Some long methods were used for parsing user input for example.

This causes low readability and makes it difficult to debug when the
project continues to scale up.

Let's split longer methods into helper functions, Javadocs can be
added, and style should be ensured for maximal readability.

In maximising readability using a universal style, others will be able
to read and understand the functionality of our code.

applepiofmyeye committed:
Merge pull request #14 from applepiofmyeye/branch-A-Assertions

Add assertions for error handling

applepiofmyeye committed:
Merge pull request #15 from applepiofmyeye/branch-A-CodeQuality

Add javadocs, change long methods, checkstyle

Commit:
Shorten longer methods

Commit:
Add basic 'View Schedules' tab

Commit:
Remove ScheduleWindow java file

Commit:
Schedule shows today's tasks

Commit:
Resolve merge conflicts in txt file

Commit:
Resolve merge conflicts A-CodeQuality branch

Commit:
Resolve merge conflicts with barbie.txt

applepiofmyeye committed:
Merge pull request #16 from applepiofmyeye/branch-A-CodeQuality

Shorten longer methods

Commit:
Merge branch 'branch-BCD-Extension'

Commit:
Merge branch-BCD-Extension with master

Commit:
Delete duplicate code

Commit:
Fix bug with launching JavaFX application

Commit:
Merge branch 'master' into branch-BCD-Extension

Commit:
Get task before accessing string: prevent duplicate getting

Commit:
Add user guide and UI image

Commit:
Edit some typos and add view feature in readme

Commit:
Add chatbot name to user guide

Commit:
Update main README.md and set up information

Commit:
Upload .jar file

Commit:
Add javadocs, remove one assertion

Commit:
Add configurations for other OSes

Commit:
Resolve jar issues, change entry point for the application

Commit:
Resolve checkstyle issues and try build gradle method

Commit:
Rebuild jar file, saved in barbie folder

Commit:
Update README

Commit:
Update README
